src/main/java/io/camunda/connector/slack/model/ConversationsInviteData.java

```
@@ -32,9 +32,14 @@ public SlackResponse invoke(MethodsClient methodsClient) throws SlackApiExceptio
     List<String> userInputAsList = null;
     if (users instanceof Collection<?>) {
       userInputAsList = (ArrayList<String>) users;
```

Will this here throw an exception if I provide a list of numbers?

---

Let's create test cases for this as well.

---

It throws an exception, but not our exception. 
It throws: class java.util.ImmutableCollections$List12 cannot be cast to class java.util.ArrayList (java.util.ImmutableCollections$List12 and java.util.ArrayList are in module java.base of loader 'bootstrap')

---

@tmetzke it will throw this error:
class java.lang.Integer cannot be cast to class java.lang.String (java.lang.Integer and java.lang.String are in module java.base of loader 'bootstrap')
Is this acceptable?

---

Two things:
- We should be as generic as possible here with the type of the user list. We could maybe cast to `Collection<String>` or at least `List<String>` instead of `ArrayList<String>`. If we test for `if (users instanceof Collection)`, we can only cast it to a collection, not a list, actually.
- Let's maybe go for something generic here that is then handled in `DataLookupService#getUserIdsFromNameOrEmail`. For example, we could simply use `Collection<?>` as the type of the `userInput` (renamed from `userInputAsList`). This can be passed on to `DataLookupService#getUserIdsFromUsers` (renamed from `getUserIdsFromNameOrEmail`). In that method, you go through the user elements and check each one's type. For those not being a String, you can either throw an error or silently ignore them.

With the proposed way, we avoid crude class cast exceptions and can take in a lot more types of data. What do you think? I can also commit a proposal, if you think that is easier. 🙂